plugins/module_utils/network/sonic/facts/sflow/sflow.py

```diff
@@ -0,0 +1,128 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2023 Dell Inc. or its subsidiaries. All Rights Reserved
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The sonic sflow fact class
+It is in this file the configuration is collected from the device
+for a given resource, parsed, and the facts tree is populated
+based on the configuration.
+"""
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from copy import deepcopy
+
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common import (
+    utils,
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.argspec.sflow.sflow import SflowArgs
+
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.sonic \
+    import to_request, edit_config
+
+
+class SflowFacts(object):
+    """ The sonic sflow fact class
+    """
+
+    def __init__(self, module, subspec='config', options='options'):
+        self._module = module
+        self.argument_spec = SflowArgs.argument_spec
+        spec = deepcopy(self.argument_spec)
+        if subspec:
+            if options:
+                facts_argument_spec = spec[subspec][options]
+            else:
+                facts_argument_spec = spec[subspec]
+        else:
+            facts_argument_spec = spec
+
+        self.generated_spec = utils.generate_dict(facts_argument_spec)
+
+    def populate_facts(self, connection, ansible_facts, data=None):
+        """ Populate the facts for sflow
+        :param connection: the device connection
+        :param ansible_facts: Facts dictionary
+        :param data: previously collected conf
+        :rtype: dictionary
+        :returns: facts
+        """
+
+        if not data:
+            data = self.get_sflow_info()
+
+        data = self.format_to_argspec(data)
+
+        # validate can add null values for things missing from device config,
+        #   so doing that before remove empties
+        cleaned_data = utils.remove_empties(
+            utils.validate_config(self.argument_spec, data)
+        )
+
+        ansible_facts['ansible_network_resources'].pop('sflow', None)
+        if cleaned_data:
+            ansible_facts['ansible_network_resources'].update({"sflow": cleaned_data["config"]})
+
+        return ansible_facts
+
+    def format_to_argspec(self, data):
+        '''takes JSON data from sflow's top level data's get REST call and returns a copy
+            that is formatted like argspec for this module. Can have empty values in it
+            :rtype: dictionary
+            :returns: dictionary that has options in same format as defined in argspec.
+            format looks something like: {"config": {"enabled":false, "interfaces":[{"name":"Etnernet0",
```

Comment:
the format here is supposed to show the returned data format. Mostly it's to add a note that it's the Argspec with the top level config key included since the rest of the code often doesn't include the config key.